src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/RemoveItemFromCartHandler.php

```diff
@@ -24,33 +25,40 @@
 /** @experimental */
 final class RemoveItemFromCartHandler implements MessageHandlerInterface
 {
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
     /** @var OrderItemRepositoryInterface */
     private $orderItemRepository;
 
     /** @var OrderModifierInterface */
     private $orderModifier;
 
     public function __construct(
+        OrderRepositoryInterface $orderRepository,
         OrderItemRepositoryInterface $orderItemRepository,
         OrderModifierInterface $orderModifier
     ) {
+        $this->orderRepository = $orderRepository;
         $this->orderItemRepository = $orderItemRepository;
         $this->orderModifier = $orderModifier;
     }
 
     public function __invoke(RemoveItemFromCart $removeItemFromCart): OrderInterface
     {
+        /** @var OrderInterface|null $cart */
+        $cart = $this->orderRepository->findOneBy(['tokenValue' => $removeItemFromCart->orderTokenValue]);
```

Comment:
```suggestion
        $cart = $this->orderRepository->findCartByTokenValue($removeItemFromCart->orderTokenValue);
```

Reply:
And I'm not sure if this database call is needed if you are using also the `orderTokenValue` to find the order item below.